bldsys/cmake/sample_helper.cmake

```diff
@@ -17,159 +17,57 @@
 
  ]]
 
-set(SCRIPT_DIR ${CMAKE_CURRENT_LIST_DIR})
+ set_property(GLOBAL PROPERTY SAMPLE_DESCRIPTORS "")
 
-function(add_sample)
+function(vkb__register_sample_descriptor)
     set(options)  
-    set(oneValueArgs ID CATEGORY AUTHOR NAME DESCRIPTION)
-    set(multiValueArgs FILES LIBS SHADER_FILES_GLSL)
+    set(oneValueArgs ID NAME DESCRIPTION LIB)
+    set(multiValueArgs)
 
     cmake_parse_arguments(TARGET "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})    
 
-    add_sample_with_tags(
-        TYPE "Sample"
-        ID ${TARGET_ID}
-        CATEGORY ${TARGET_CATEGORY}
-        AUTHOR ${TARGET_AUTHOR}
-        NAME ${TARGET_NAME}
-        DESCRIPTION ${TARGET_DESCRIPTION}
-        TAGS 
-            "any"
-        FILES
-            ${SRC_FILES}
-        LIBS
-            ${TARGET_LIBS}
-        SHADER_FILES_GLSL
-            ${TARGET_SHADER_FILES_GLSL})
+    get_property(DESCRIPTORS GLOBAL PROPERTY SAMPLE_DESCRIPTORS)
+    list(APPEND DESCRIPTORS "{\"id\":\"${TARGET_ID}\",\"name\":\"${TARGET_NAME}\", \"description\": \"${TARGET_DESCRIPTION}\", \"library_name\": \"${TARGET_LIB}\"}")
```

Comment:
could potentially use jq or https://github.com/sbellus/json-cmake or some such to make this a usable thing at the cmake level. however, this works.

Reply:
Does this allow for creating json structs in cmake? Or just parsing?